xtl fails to build without -fpermissive

There is a small issue with xtl xany.

In particular,

```
xany.hpp:394:31: error: invalid conversion from 'const void*' to 'void*' [-fpermissive]                                                  │
         detail::check_any_cast(p);                                                                                                                                                  │
         ~~~~~~~~~~~~~~~~~~~~~~^~~

```
Changing the parameter's type from `void *p` in this line 

https://github.com/QuantStack/xtl/blob/d9cc8938ebdd46cbda4fe619715a2cf92905c754/include/xtl/xany.hpp#L40

to `const void *p` fixes the issue.

I won't submit a patch because it's just a quick fix that can be done manually.

Recreate the environment as follows:

System: Ubuntu 18.10 (clean)
System GCC: gcc (Ubuntu 8.2.0-7ubuntu1) 8.2.0
System CMAKE: cmake version 3.12.1 (apt installed)

Starting with a clean install of conda  (miniconda)

conda --version: conda 4.6.11

Create a new conda environment
```
conda create -n xwidgets-tests
conda activate xwidgets-tests
```

Install the following:

```
conda install -n xwidgets-tests -c conda-forge cling xwidgets
```

My 
`conda env export` is:

```
name: xwidget-tests
channels:
  - conda-forge
  - defaults
dependencies:
  - binutils_impl_linux-64=2.31.1=h6176602_1
  - binutils_linux-64=2.31.1=h6176602_3
  - ca-certificates=2019.3.9=hecc5488_0
  - clang_variant=1.0=cling_6.14.06
  - clangdev=5.0.0=h935a590_1004
  - cling=0.5=he860b03_1006
  - cppzmq=4.3.0=h484f380_1000
  - gcc_impl_linux-64=7.3.0=habb00fd_1
  - gcc_linux-64=7.3.0=h553295d_3
  - gxx_impl_linux-64=7.3.0=hdf63c60_1
  - gxx_linux-64=7.3.0=h553295d_3
  - libgcc-ng=8.2.0=hdf63c60_1
  - libsodium=1.0.16=h14c3975_1001
  - libstdcxx-ng=8.2.0=hdf63c60_1
  - libuuid=2.32.1=h14c3975_1000
  - nlohmann_json=3.5.0=hf484d3e_0
  - openssl=1.1.1b=h14c3975_1
  - xeus=0.19.1=h4d8c418_1
  - xproperty=0.8.1=hc9558a2_1
  - xtl=0.6.2=hc9558a2_0
  - xwidgets=0.17.0=hc9558a2_0
  - zeromq=4.2.5=hf484d3e_1006
  - zlib=1.2.11=h14c3975_1004
prefix: /home/masud/.local/miniconda3/envs/xwidget-tests

```

ldd $(which cling) to verify that the linked libraries for libgcc are the ones in conda environment.

At command line launch `cling -std=c++14 --nologo`

at the [cling]$ prompt type `.I` to verify the include paths.

then at the [cling]$ prompt type:

```
#include "xwidgets/xslider.hpp"
xw::slider<double> slider;
```

This should result in the following:
```
[cling]$ #include "xwidgets/xslider.hpp"
[cling]$ xw::slider <double> slider;
In file included from input_line_3:1:
In file included from //...//miniconda3/envs/xwidget-tests/include/xwidgets/xslider.hpp:19:
In file included from //...//miniconda3/envs/xwidget-tests/include/xwidgets/xnumber.hpp:12:
In file included from //...//miniconda3/envs/xwidget-tests/include/xwidgets/xwidget.hpp:15:
In file included from //...//miniconda3/envs/xwidget-tests/include/xwidgets/xlayout.hpp:16:
In file included from //...//miniconda3/envs/xwidget-tests/include/xwidgets/xobject.hpp:18:
In file included from //...//miniconda3/envs/xwidget-tests/include/xproperty/xobserved.hpp:18:
//...//miniconda3/envs/xwidget-tests/include/xtl/xany.hpp:394:32: error: cannot initialize a parameter of type 'void *' with an lvalue of type 'const std::function<void (xw::xmaterialize<xslider, double> &,
      xtl::xoptional<std::__cxx11::basic_string<char>, bool> &)> *'
        detail::check_any_cast(p);
                               ^

[---snipped---]
```
and a whole bunch of other errors... ending in a segfault.

Now.. after making the proposed change to xtl/xany.hpp

And repeating the cling steps 

We get the desired result :) of things just working fine as expected.

Hope this helps to recreate the issue.

Cheers, 

Ahmed

--- a/include/xtl/xany.hpp
+++ b/include/xtl/xany.hpp
@@ -37,7 +37,7 @@ namespace xtl
     };
 
     namespace detail {
-        inline static void check_any_cast(void* p) {
+        inline static void check_any_cast(const void* p) {
             if (p == nullptr) {
 #if defined(XTL_NO_EXCEPTIONS)
                 std::fprintf(stderr, "bad_any_cast\n");

